x-pack/docs/en/security/authentication/configuring-kerberos-realm.asciidoc

```diff
@@ -0,0 +1,182 @@
+mappings in your domain. Refer to your Kerberos documentation for more details.
+
+These instructions do not cover setting up and configuring your Kerberos 
+deployment. Where examples are provided, they pertain to an MIT Kerberos V5 
+deployment. For more information, see 
+http://web.mit.edu/kerberos/www/index.html[MIT Kerberos documentation]
+--
+
+. Configure Java GSS. 
++
+--
+
+{es} uses Java GSS framework support for Kerberos authentication.
+To support Kerberos authentication, {es} needs the following files:
+
+* `krb5.conf`, a Kerberos configuration file
+*  A `keytab` file that contains credentials for the {es} service principal
+
+The configuration requirements depend on your Kerberos setup. Refer to your 
+Kerberos documentation to configure the `krb5.conf` file.
+
+For more information on Java GSS, see 
+https://docs.oracle.com/javase/10/security/kerberos-requirements1.htm[Java GSS Kerberos requirements]
+--
+
+==== Create a Kerberos realm
+
+To configure a Kerberos realm in {es}:
+
+. Enable SSL/TLS for HTTP. 
++
+--
+
+If your {es} cluster is operating in production mode, then you must
+configure the HTTP interface to use SSL/TLS before you can enable
+Kerberos authentication.
+
+For more information, see
+{ref}/configuring-tls.html#tls-http[Encrypting HTTP Client Communications].
+
+--
+
+. Configure the JVM to find the Kerberos configuration file. 
++
+--
+{es} uses Java GSS and JAAS Krb5LoginModule to support Kerberos authentication 
+using a Simple and Protected GSSAPI Negotiation Mechanism (SPNEGO) mechanism. 
+The Kerberos configuration file (`krb5.conf`) provides information such as the 
+default realm, the Key Distribution Center (KDC), and other configuration details 
+required for Kerberos authentication. When the JVM needs some configuration 
+properties, it tries to find those values by locating and loading this file. The 
+JVM system property to configure the file path is `java.security.krb5.conf`. To 
+configure JVM system properties see {ref}/jvm-options.html[configuring jvm options]. 
+If this system property is not specified, Java tries to locate the file based on 
+the conventions.
+
+TIP: It is recommended that this system property be configured for {es}.
+The method for setting this property depends on your Kerberos infrastructure. 
+Refer to your Kerberos documentation for more details.
+
+For more information, see http://web.mit.edu/kerberos/krb5-latest/doc/admin/conf_files/krb5_conf.html[krb5.conf]
+
+--
+
+. Create a keytab for the service node. 
++
+--
+A keytab is a file that stores pairs of principals and encryption keys. {es} 
+uses the key from the keytab to decrypt the tickets presented by the user. You 
+must create a keytab for {es} by using the tools provided by your Kerberos 
+implementation. For example, some tools that create keytabs are `ktpass.exe` on 
+Windows and `kadmin` for MIT Kerberos. 
+--
+
+. Put the keytab file in the `ES_PATH_CONF`.
```

Comment:
Per earlier comment, can we put it in a different path if keytab.path is set appropriately?